vars/gangplank.groovy

```diff
@@ -103,6 +103,60 @@ def runSpec(params =[:]) {
         gangplankCmd = _getMode(params)
         gangplankCmd += "--spec ${params['spec']} "
         runGangplank(gangplankCmd)
+    } else {
+        error("runSpec requires a spec param.")
+    }
+}
+
+// Available parameters:
+//     cmd:           string  -- the single command to run
+//     mode:          string  -- Gangplank Mode
+//     extraFlags:    string  -- Extra flags to use
+def runSingleCmd(params = [:]) {
+    if (params['cmd']) {
+        gangplankCmd = _getMode(params)
+        gangplankCmd += " --singleCmd \"${params['cmd']}\""
+        runGangplank(gangplankCmd)
+    } else {
+        error("runSingleCmd requires a cmd param.")
+    }
+}
+
+// A set of default parameters for building our FCOS AARCH64 builds via gangplank.
+// - use a locally build COSA because we don't currently build/push multi-arch COSA anywhere. 
+// - use pod mode with --podman for remote podman execution
+// - use the 'builds' "bucket" will fetch artifacts back into the expected 'builds' directory
+// - target aarch64 architecture
+def getGangPlankFCOSAARCH64Params() {
+    return [mode: 'pod',
+            extraFlags: '--podman --bucket=builds --arch=aarch64',
+            image: 'localhost/coreos-assembler:latest']
+}
+
+// A function to wrap what's needed to run gangplank for our aarch64
+// builder host. Accepts a params map with the usual parameters for
+// a call do runSpec or runSingleCmd.
+def gangPlankFCOSAARCH64BuilderWrapper(params = [:]) {
+    withCredentials([
+        string(credentialsId: 'fcos-aarch64-builder-host-string',
+               variable: 'REMOTEHOST'),
+        string(credentialsId: 'fcos-aarch64-builder-uid-string',
+               variable: 'REMOTEUID'),
+        sshUserPrivateKey(credentialsId: 'fcos-aarch64-builder-sshkey-key',
+                          usernameVariable: 'REMOTEUSER',
+                          keyFileVariable: 'CONTAINER_SSHKEY')
+    ]) {
```

Comment:
Hmm, this seems odd in the library. It's specific to the pipeline. I'd personally prefer having it live in the pipeline as a separate function (and then here we can just pass in the user, host, and uid bits). At the very least, let's make them parameters instead of hardcoding the ID strings.

Reply:
moved bits to the pipeline in `gp.groovy`.